### PR TITLE
UI: Right sidebar UI

### PR DESCRIPTION
**Pull request by jordanvidrine**
This PR brings the right sidebar injected component up to date with the topic-post event rendering. (when the right sidebar blocks theme component is installed & active)

Related PR -> https://github.com/discourse/discourse-calendar/pull/641

After
image

Before
image

### CODE REVIEW

File: assets/javascripts/discourse/components/upcoming-events-list.gjs [hunk 1]
```diff
@@ -48,14 +44,21 @@ export default class UpcomingEventsList extends Component {
...
     super(...arguments);
-
     this.appEvents.on("page:changed", this, this.updateEventsList);
   }
 
   get categoryId() {
     return this.router.currentRoute.attributes?.category?.id;
   }
 
+  get hasEmptyResponse() {
```
Comment (brrusselburg) [inline assets/javascripts/discourse/components/upcoming-events-list.gjs:54]: is there a reason this got moved around?

Comment (jordanvidrine) [reply assets/javascripts/discourse/components/upcoming-events-list.gjs:54]: a rebase issue

**Comment by brrusselburg** [comment]
Here's what I'm seeing (local settings remove time format):
Screenshot 2024-12-05 at 4 24 53 PM

Looks pretty good to me. Not sure if you want any styling changes in case time is removed.

Messed around with some settings to see if anything would break for people who have the sidebar calendar set up as currently instructed. But all good! 🕺 🪩

**Comment by brrusselburg** [comment]
I think there are some areas for refactor but they have nothing to do with design / the work you're doing. Areas where the code refers to something slightly different than it did before these changes, that's all. I'll make a note-to-self.

**Review by brrusselburg — APPROVED** [review]

**Pull request merged**